lib/Sema/ConstraintSystem.h

```diff
@@ -1622,6 +1634,19 @@ class ConstraintSystem {
     return TypeLocTypes.find(&L)->second;
   }
 
+  Type getType(const ValueDecl *P, bool queryDeclOnMiss = true) const {
```

Comment:
Can I suggest splitting this into three functions?
  `getType(const ParamDecl *)` - asserts hasType() and returns the type

  `getTypeOrNull(const ValueDecl *D)`:
```
      if (auto *P = dyn_cast<ParamDecl>(D)) 
        return getType(P);

      return Type();
```

`getTypeOrInterfaceType`: Your current function, minus the argument for `queryDeclOnMiss`

Reply:
Sure!